staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -282,11 +285,22 @@ func startControllers(cloud cloudprovider.Interface, ctx genericcontrollermanage
 			klog.Warningf("Skipping %q", controllerName)
 			continue
 		}
+		check := controllerhealthz.NamedPingChecker(controllerName)
+		if ctrl != nil {
+			if healthCheckable, ok := ctrl.(controller.HealthCheckable); ok {
+				if realCheck := healthCheckable.HealthChecker(); realCheck != nil {
+					check = realCheck
```

Comment:
Could two controllers provide health checks with the same name? What would happen?

Reply:
The standard `healthz` mechanism disallows that. It will panic when two checks try to mount on the same path. e.g. two checks named "foo", first mounts on "/healthz/foo", second mounts on the same path, which will panic (from implementation of mux).

Reply:
Fortunately, if two controllers provide the same health checks, the controller manager would panic on start, catching the error during any e2e test.

Reply:
That's good. Would it possible for us to validate the names in the provider checks match the controller name? If not, what we have might be okay, but if we enforce the naming that would be great.

Reply:
I think there is another way. We won't let the controller decide the name (with a new interface instead of healthz.HealthChecker). The controller manager decide the name of a check.

Reply:
Updated.

Reply:
Excellent.